web/src/beta/lib/core/engines/Cesium/pickMany.ts

```diff
@@ -0,0 +1,302 @@
+import { SceneTransforms, type Scene } from "@cesium/engine";
+import Cesium3DTilePass from "@cesium/engine/Source/Scene/Cesium3DTilePass";
+import Cesium3DTilePassState from "@cesium/engine/Source/Scene/Cesium3DTilePassState";
+import {
+  BoundingRectangle,
+  Cartesian2,
+  Color,
+  PerspectiveFrustum,
+  PerspectiveOffCenterFrustum,
+  type CullingVolume,
+  Viewer,
+} from "cesium";
+import invariant from "tiny-invariant";
+
+import { ComputedFeature } from "../../Map";
+import { assertType } from "../../utils";
+
+import { convertObjToComputedFeature, getPixelRatio } from "./utils";
+
+declare module "@cesium/engine" {
+  // eslint-disable-next-line @typescript-eslint/no-namespace
+  namespace SceneTransforms {
+    function transformWindowToDrawingBuffer(
+      scene: Scene,
+      windowPosition: Cartesian2,
+      result?: Cartesian2,
+    ): Cartesian2;
+  }
+}
+
+interface ReadState {
+  x: number;
+  y: number;
+  width: number;
+  height: number;
+  framebuffer: unknown;
+}
+
+interface PickFramebuffer {
+  _fb: {
+    framebuffer: unknown;
+  };
+  _context: {
+    readPixels: (readState: ReadState) => Uint8Array;
+    getObjectByPickColor: (color: Color) => unknown | undefined;
+  };
+  begin: (screenSpaceRectangle: BoundingRectangle, viewport: BoundingRectangle) => PassState;
+  end: (screenSpaceRectangle: BoundingRectangle) => unknown | undefined;
+}
+
+interface PassState {
+  viewport: BoundingRectangle;
+}
+
+interface FrameState {
+  passes: {
+    pick: boolean;
+  };
+  cullingVolume?: CullingVolume;
+  invertClassification: boolean;
+  tilesetPassState?: PassState;
+}
+
+interface UniformState {
+  update: (frameState: FrameState) => void;
+}
+
+interface Context {
+  uniformState: UniformState;
+  drawingBufferWidth: number;
+  drawingBufferHeight: number;
+  endFrame: () => void;
+}
+
+interface JobScheduler {
+  disableThisFrame: () => void;
+}
+
+interface View {
+  viewport: BoundingRectangle;
+  passState: PassState;
+  pickFramebuffer: PickFramebuffer;
+}
+
+interface PrivateScene extends Scene {
+  view: View;
+  defaultView: View;
+  jobScheduler: JobScheduler;
+  frameState: FrameState;
+  context: Context;
+  updateFrameState: () => void;
+  updateEnvironment: () => void;
+  updateAndExecuteCommands: (passState: PassState, backgroundColor: Color) => void;
+  resolveFramebuffers: (passState: PassState) => void;
+}
+
+const colorScratch = new Color();
+
+// Derived from: https://github.com/CesiumGS/cesium/blob/1.105/packages/engine/Source/Scene/PickFramebuffer.js#L54
+// Picks multiple objects in the given rectangle.
+function pickFramebufferEnd(
+  this: PickFramebuffer,
+  screenSpaceRectangle: BoundingRectangle,
+): object[] {
+  const width = screenSpaceRectangle.width ?? 1;
+  const height = screenSpaceRectangle.height ?? 1;
+  const context = this._context;
+  const pixels = context.readPixels({
```

Comment:
Now this one is optional, 
You might want to add validations for the returned `pixels` and handle potential issues gracefully, to avoid unexpected behavior in subsequent lines of code.